### ISSUE TITLE
Support for type checking fixed width ints

### ISSUE THREAD

**Issue by adsharma** [microsoft/pyright#1872]
**Is your feature request related to a problem? Please describe.**

Working on transpiling python3 to a host of other statically typed languages. This effort can make use of fixed width integer types in python.

Currently the implementation uses ctypes. Sample code:

```
#!/usr/bin/env python3

from ctypes import c_int8 as i8

def test():
    a: i8 = 10
    b: i8 = 20
    c = a + b
    reveal_type(c)
```

**Describe the solution you'd like**

```
a: i8 = 300
```

should produce a type error that says 300 doesn't fit in an i8. Right now I get:

```
Expression of type "Literal[300]" cannot be assigned to declared type "c_int8"
```

Also, there is a question on what should reveal_type(c) should print. Some languages prefer i8, other languages prefer int.

I'm trying to define new semantics that assists with overflow/underflow safe programming. Any additional support from pyright in catching such errors before transpilation would be great.

**Additional context**

https://github.com/adsharma/py2many/issues/123

**Comment by erictraut** [comment]
The Python type system has no way of representing numeric widths or ranges. Pyright supports the typing standards spelled out in various PEPs. If you want to see extensions to the type system, you're welcome to post in the typing-sig discussions and propose new mechanisms. We don't currently have any plans to pursue such extensions given that this is a very niche use case in Python.

**Comment by adsharma** [comment]
There is some support for doing this in my fork of typpete, a z3 based type checker.

https://github.com/adsharma/Typpete/commit/256fdce48d80d04497a4f7bba9ab280810367f94
